pkg/frontend/querymiddleware/codec.go

```diff
@@ -238,7 +213,8 @@ func newPrometheusCodecMetrics(registerer prometheus.Registerer) *prometheusCode
 	}
 }
 
-type prometheusCodec struct {
+// PrometheusCodec is used to encode/decode query requests and responses so they can be passed down to middlewares.
+type PrometheusCodec struct {
```

Comment:
How about we just name this `Codec`? The `Prometheus` prefix makes me think this is an implementation of some interface, which it isn't anymore.

Reply:
Good point. I agree `Codec` is just fine. I will do it.